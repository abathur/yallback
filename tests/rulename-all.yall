# - `yallback:rule:<rulename>:all` - receive all rule matches in a single call
#     - `$1` == `<rulename>` (this is a convenience, in case you want a single function to handle multiple rules)
#     - matching files are passed on stdin, one per line

yallback:rule:binary:all(){
	if ! expect_rulename_all "$1" "binary"; then
		fail "unexpected output"
	fi
}

yallback:rule:fake:all(){
	if ! expect_rulename_all "$1" "fake"; then
		fail "unexpected output"
	fi
}

yallback:rule:go_binary:all(){
	if ! expect_rulename_all "$1" "go_binary"; then
		fail "unexpected output"
	fi
}

yallback:rule:go_exec:all(){
	if ! expect_rulename_all "$1" "go_exec"; then
		fail "unexpected output"
	fi
}
# set -x
alias yallback:rule:decidable:all=true
