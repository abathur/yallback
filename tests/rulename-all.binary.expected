/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/addr2line
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/api
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/asm
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/authtest
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/benchcmp
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/buildid
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/bundle
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/callgraph
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/cgo
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/compile
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/compilebench
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/cookieauth
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/cover
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/digraph
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/dist
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/doc
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/eg
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/fieldalignment
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/findcall
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/fix
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/fiximports
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/getgo
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gitauth
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/go-contrib-init
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/godex
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/godoc
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/goimports
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gomvpkg
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gopackages
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gorename
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gostacks
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/gotype
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/goyacc
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/guru
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/helper
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/html2article
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/ifaceassert
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/link
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/lostcancel
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/netrcauth
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/nilness
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/nm
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/objdump
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/pack
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/pprof
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/present
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/present2md
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/server
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/shadow
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/splitdwarf
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/ssadump
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/stress
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/stringer
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/stringintconv
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/test2json
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/toolstash
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/trace
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/unmarshal
/nix/store/qnj9dck4kbv5dwmd0z8p0mbbmph7jjjr-gotools-unstable-2021-01-13/bin/vet
